overlord/devicestate/devicestate_bootconfig_test.go

```diff
@@ -290,16 +376,62 @@ func (s *deviceMgrBootconfigSuite) TestBootConfigUpdateWithGadgetExtra(c *C) {
 	})
 }
 
+func (s *deviceMgrBootconfigSuite) TestBootConfigUpdateRunWithAppend(c *C) {
+	s.state.Lock()
+	s.setupUC20Model(c)
+	s.state.Unlock()
+
+	s.managedbl.Updated = true
+
+	opts := testBootConfigUpdateOpts{
+		updateAttempted: true,
+		updateApplied:   true,
+		cmdlineAppend:   "par1=val par2",
+	}
+	s.testBootConfigUpdateRun(c, opts, "")
+
+	m, err := boot.ReadModeenv("")
+	c.Assert(err, IsNil)
+	c.Check([]string(m.CurrentKernelCommandLines), DeepEquals, []string{
+		"snapd_recovery_mode=run console=ttyS0 console=tty1 panic=-1",
+		"snapd_recovery_mode=run console=ttyS0 console=tty1 panic=-1 candidate par1=val par2",
+	})
+}
+
+func (s *deviceMgrBootconfigSuite) TestBootConfigUpdateRunWithDangerousAppend(c *C) {
+	s.state.Lock()
+	s.setupUC20Model(c)
+	s.state.Unlock()
+
+	s.managedbl.Updated = true
+
+	opts := testBootConfigUpdateOpts{
+		updateAttempted:     true,
+		updateApplied:       true,
+		cmdlineAppendDanger: "par1=val par2",
```

Comment:
should we have a test where both cmdlineAppend and cmdlineAppendDanger are set?

Reply:
do we have tests here where the grade is not dangerous?

Reply:
Right, added tests for that too